dangerfile.ts

```diff
@@ -161,3 +162,23 @@ if (pr.body.replace(neuterMarkdownTicks, "-") !== newBody.replace(neuterMarkdown
   // See https://github.com/artsy/emission/issues/589
   // danger.github.api.pullRequests.update({...danger.github.thisPR, body: newBody })
 }
+
+// No `yarn run` inside the Package.json
+// It would make the shell glob instead of CLI tools which do a better job.
+// See: https://github.com/yarnpkg/yarn/issues/3595
+const packageText = fs.readFileSync("package.json", "utf8")
+if (packageText.includes("yarn run")) {
+  const url = "https://github.com/yarnpkg/yarn/issues/3595"
+  fail(`You have a \`yarn run\` inside the package.json. This is probably a mistake, see ${url}.`)
+}
```

Comment:
_strictly speaking_ this would not have caught this issue, I couldn't think of a way to detect the implicit `yarn run` but it definitely protects against it in the future